### PR TITLE
Skip pages with anchors and discography items

### PR DESCRIPTION
**Pull request by mineo**
The individual commits have more comments.

It would also be good to let @reosarevok have a look at this.

### CODE REVIEW

File: bot/common.py [hunk 1]
```diff
@@ -43,8 +43,34 @@ def signal_handler(signal, frame):
...
-class IsDisambigPage(Exception):
-    pass
+class SkipPage(Exception):
+    def __init__(self, url):
+        self.url = url
+
+
+class IsDisambigPage(SkipPage):
+    def __str__(self):
+        return "{url} is a disambigutation page".format(url=self.url)
```
Comment (zas) [inline bot/common.py]: disambiguation

File: bot/common.py [hunk 2]
```diff
@@ -155,10 +158,24 @@ def do_readwrite_query(query, vars=None):
     return cur
 
 
+def check_has_fragment(url):
+    """Check if `url` contains a fragment
+
+    This is most often the case for discography pages where a single album is
+    only mentioned in a few paragraphs."""
+    parsed_url = urlparse(url)
+    if parsed_url.fragement:
```
Comment (reosarevok) [inline bot/common.py]: fragement?

File: bot/common.py [hunk 3]
```diff
@@ -58,6 +58,11 @@ def __str__(self):
         return "{url} has a fragment".format(url=self.url)
 
 
+class IsRedirectWithItemPage(SkipPage):
+    def __str__(self):
+        return "{url} is a redirect page, but is linked to a Wkidata item".format(url=self.url)
```
Comment (reosarevok) [inline bot/common.py]: "Wkidata"

File: bot/common.py [hunk 4]
```diff
@@ -173,6 +178,22 @@ def check_url_needs_to_be_skipped(wikilink, page):
...
+            # Examples of this are
+            # https://en.wikipedia.org/w/index.php?title=Star_Wars_Episode_I:_The_Phantom_Menace_(soundtrack)&redirect=no
+            pass
+        else:
+            # Page is a redirect, but has its own wikidata item - it's better
+            # to not touch this.
+            # Examples of this are
+            # https://en.wikipedia.org/wiki/Hybrid_Theory_%28EP%29, which is a
+            # redirect on en.wp, but has its own article on lots of other
+            # wikipedias and therefore its own wikidata item.
```
Comment (reosarevok) [inline bot/common.py:205]: If possible, if the Wikipedia page matches a Wikidata item already present on the entity, we could just remove the unneeded Wikipedia page, too.

Comment (mineo) [reply bot/common.py:205]: Unfortunately, the `musicbrainz_bot` module used for editing only supports adding or editing URLs, but not removing them :-(

**Review by zas — CHANGES_REQUESTED** [review]

**Comment by mineo** [comment]
If there are no further comments, can you merge and deploy this, @reosarevok?

**Pull request merged**